pkg/devfile/adapters/docker/component/adapter.go

```diff
@@ -74,20 +75,48 @@ func (a Adapter) Delete(labels map[string]string) error {
 		return errors.Wrap(err, "unable to retrieve container list for delete operation")
 	}
 
+	// A unique list of volumes NOT to delete, because they are still mapped into other containers.
+	// map key is volume name.
+	volumesNotToDelete := map[string]string{}
+
+	// Go through the containers which are NOT part of this component, and make a list of all
+	// their volumes so we don't delete them.
+	for _, container := range list {
```

Comment:
Why not collect the list of unique volumes while deleting each container, then once all of the containers have been removed, removed the unique voluems?